### ISSUE TITLE
Set default learning rate for t-SNE to "auto"

### ISSUE THREAD

**Issue by huddlej** [blab/pathogen-embed#4]
Based on our results from applying t-SNE to simulated populations in https://github.com/blab/cartography, a more reasonable default learning rate should scale with the input sample size instead of being fixed at 200.0. [The default learning rate in the scikit-learn implementation we use is already the string "auto"](https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html#sklearn.manifold.TSNE) which allows the learning rate to scale with the sample size. We should change [the default value for the `tsne` subcommand's `--learning-rate` argument](https://github.com/blab/pathogen-embed/blob/5d9c64e76a78ebbe209714046d5750a59f51746d/src/pathogen_embed/__main__.py#L30) to `"auto"`. This change will require us to define [a custom argparse type function](https://docs.python.org/3/library/argparse.html#type), since valid values can be either a `float` or a string with the value of `auto`.
